Makefile.am

```diff
@@ -51,13 +51,68 @@ genfiles/admin_lua_proc.h: lua/admin.lua
+nodist_libvalhalla_la_SOURCES = genfiles/date_time_zonespec.h genfiles/graph_lua_proc.h genfiles/admin_lua_proc.h genfiles/locales.h
+CLEANFILES = $(EXTRA_PROGRAMS) genfiles/date_time_zonespec.h genfiles/graph_lua_proc.h genfiles/admin_lua_proc.h genfiles/locales.h valhalla/proto/tripcommon.pb.h valhalla/proto/trippath.pb.h valhalla/proto/directions_options.pb.h valhalla/proto/tripdirections.pb.h src/proto/tripcommon.pb.cc src/proto/trippath.pb.cc src/proto/directions_options.pb.cc src/proto/tripdirections.pb.cc
+
+#data tools protobuffs
+if DATA_TOOLS
+valhalla/proto/osmformat.pb.h: src/proto/osmformat.pb.cc
+	-mkdir -p valhalla/proto && mv src/proto/osmformat.pb.h valhalla/proto
+src/proto/osmformat.pb.cc:
+	@echo " PROTOC osmformat.proto" && mkdir -p src/proto && @PROTOC_BIN@ -Iproto --cpp_out=src/proto proto/osmformat.proto
+valhalla/proto/fileformat.pb.h: src/proto/fileformat.pb.cc
+	-mkdir -p valhalla/proto && mv src/proto/fileformat.pb.h valhalla/proto
+src/proto/fileformat.pb.cc:
+	@echo " PROTOC fileformat.proto" && mkdir -p src/proto && @PROTOC_BIN@ -Iproto --cpp_out=src/proto proto/fileformat.proto
+valhalla/proto/transit.pb.h: src/proto/transit.pb.cc
+	-mkdir -p valhalla/proto && mv src/proto/transit.pb.h valhalla/proto
+src/proto/transit.pb.cc:
+	@echo " PROTOC transit.proto" && mkdir -p src/proto && @PROTOC_BIN@ -Iproto --cpp_out=src/proto proto/transit.proto
+valhalla/proto/tile.pb.h: src/proto/tile.pb.cc
+	-mkdir -p valhalla/proto && mv src/proto/tile.pb.h valhalla/proto
+src/proto/tile.pb.cc:
+	@echo " PROTOC tile.proto" && mkdir -p src/proto && @PROTOC_BIN@ -Iproto --cpp_out=src/proto proto/tile.proto
+valhalla/proto/segment.pb.h: src/proto/segment.pb.cc
+	-mkdir -p valhalla/proto && mv src/proto/segment.pb.h valhalla/proto
+src/proto/segment.pb.cc:
+	@echo " PROTOC segment.proto" && mkdir -p src/proto && @PROTOC_BIN@ -Iproto --cpp_out=src/proto proto/segment.proto
+BUILT_SOURCES += valhalla/proto/osmformat.pb.h valhalla/proto/fileformat.pb.h valhalla/proto/transit.pb.h valhalla/proto/tile.pb.h valhalla/proto/segment.pb.h
+nodist_libvalhalla_la_SOURCES += valhalla/proto/osmformat.pb.h valhalla/proto/fileformat.pb.h valhalla/proto/transit.pb.h valhalla/proto/tile.pb.h valhalla/proto/segment.pb.h
+CLEANFILES += valhalla/proto/osmformat.pb.h valhalla/proto/fileformat.pb.h valhalla/proto/transit.pb.h valhalla/proto/tile.pb.h valhalla/proto/segment.pb.h src/proto/tripcommon.pb.cc src/proto/trippath.pb.cc src/proto/directions_options.pb.cc src/proto/tripdirections.pb.cc src/proto/osmformat.pb.cc src/proto/fileformat.pb.cc src/proto/transit.pb.cc src/proto/tile.pb.cc src/proto/segment.pb.cc
+endif
+
+#embedded device protobuffs
+#if EMBEDDED
```

Comment:
note that we dont actually have control over this yet so its currently not optional